### PR TITLE
WIP: try spj

### PR DESCRIPTION
**Pull request by twd2**
TODO: dataset format

### CODE REVIEW

File: jd4/case.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import pyximport; pyximport.install()
```
Comment (iceboy233) [inline jd4/case.py:1]: please don't use this in final version, as it increase startup time and memory footprint (it's also essential to keep the memory footprint small before forking workers). use static compilation (see setup.py). and why do you need to access cython module here?

File: jd4/spj.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
+import pyximport; pyximport.install()
```
Comment (iceboy233) [inline jd4/spj.py:1]: please don't duplicate code. prefer to add these into case.py . also could you create a "modern case" (might need to think about the naming), correspond to the "legacy case", and make spj optional (to support configuration)?

File: jd4/spj.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    async def judge(self, judge_sandbox, user_sandbox, judge_package, user_package):
+        loop = get_event_loop()
+        judge_exe = await judge_package.install(judge_sandbox)
+        user_exe = await user_package.install(user_sandbox)
+        judge_stdin_file = path.join(judge_sandbox.in_dir, 'stdin')
+        mkfifo(judge_stdin_file)
+        judge_stdout_file = path.join(judge_sandbox.in_dir, 'stdout')
+        mkfifo(judge_stdout_file)
+        # TODO(twd2): performance improve
```
Comment (twd2) [inline jd4/spj.py:57]: @iceb0y PTAL

File: jd4/spj.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                            JUDGE_MEM_KB * 1024,
+                            JUDGE_PROCESS_LIMIT),
+                wait_cgroup(user_cgroup_sock,
+                            user_exe_task,
+                            self.time_limit_ns,
+                            self.memory_limit_bytes,
+                            self.process_limit))
+            judge_exe_status = await judge_exe_task
+            user_exe_status = await user_exe_task
+            _, __, judge_err, user_err, (judge_time_usage_ns, judge_memory_usage_bytes), \
```
Comment (iceboy233) [inline jd4/spj.py:101]: fyi you can use `_, _, ...`

**Comment by twd2** [comment]
Should be replaced by #57.

**Pull request closed without merging**